invader_payment_adyen_klarna/services/payment_adyen.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2019 ACSONE SA/NV (http://acsone.eu).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo.fields import first
+
+from odoo.addons.component.core import AbstractComponent
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentServiceAdyen(AbstractComponent):
+    _inherit = "payment.service.adyen"
+
+    def _klarna_partner_info(self, partner):
+        return {
+            "city": partner.city,
+            "country": partner.country_id.code,
+            "houseNumberOrName": "",
+            "postalCode": partner.zip,
+            "street": partner.street,
```

Comment:
@acsonefho Be careful since none of these fields are mandatory into odoo. Therefore, you must avoid to return False in place of an empty string...  `"city": partner.city or ""`